dds/DCPS/ReactorInterceptor.h

```diff
@@ -35,14 +35,15 @@ class OpenDDS_Dcps_Export ReactorInterceptor : public RcEventHandler {
   typedef container_supported_unique_ptr<Command> CommandPtr;
 
   bool should_execute_immediately();
-  void process_command_queue();
 
   template<typename T>
   void execute_or_enqueue(T& t)
   {
     if (should_execute_immediately()) {
-      ACE_GUARD(ACE_Thread_Mutex, guard, this->mutex_);
-      process_command_queue();
+      {
```

Comment:
From this code, I'm inferring that you want strict FIFO ordering so that anything that was deferred is executed first.  This makes sense.

Reply:
I just needed the ReactorIntercepter's mutex to be unlocked during the call to Command::execute, since some commands may need to call back into the ReactorInterceptor. I think the FIFO behavior remains unchanged here, and I agree that it makes sense.